### PR TITLE
docs/dev: add a quickstart guide

### PR DESCRIPTION
**Pull request by lucab**
This adds a quickstart guide for developers, showing how to build/test
Zincati and how to embed it into an OS image via coreos-assembler.

Closes: https://github.com/coreos/zincati/issues/353

### CODE REVIEW

File: docs/development/quickstart.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Development quickstart
```
Comment (travier) [inline docs/development/quickstart.md:1]: Can you include:
```
---
layout: default
nav_order: 1
parent: Development
---
```
And probably update the order of the other pages?

**Review by travier — APPROVED** [review]
LGTM with a small suggestion.

File: docs/development/quickstart.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+```sh
+make build
+make check
+```
+
+If you prefer running builds in a containerized environment, you can use the provided `Dockerfile`:
+
+```sh
+docker build -f contrib/Dockerfile.dev -t coreos/zincati:dev .
```
Comment (jlebon) [inline docs/development/quickstart.md:44]: There's also the cosa buildroot image which has cosa + build deps we could link to here. This is actually what CI here itself uses: https://github.com/coreos/zincati/blob/9d5d5ebe941151aff1d1b5efbbb816553de94f9b/.cci.jenkinsfile#L3 -> https://github.com/coreos/coreos-ci-lib/blob/76bbd3d46503dea956c59eb02a079302de0d3abb/vars/cosaPod.groovy#L8.

Nice to have a minimal image as well though given that the buildroot image is massive.

File: docs/development/quickstart.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+mkdir test-image
+cd test-image
+cosa init https://github.com/coreos/fedora-coreos-config
+popd
+docker build -f contrib/Dockerfile.dev -t coreos/zincati:dev .
+docker run --rm -v "$(pwd):/source" -v "/tmp/test-image:/assembler" coreos/zincati:dev
+pushd /tmp/test-image
+cosa fetch
+cosa build
+```
```
Comment (jlebon) [inline docs/development/quickstart.md:65]: Could we link to https://coreos.github.io/coreos-assembler/working/#using-overrides here as well for the "pet container" case where cosa and build tools are in the same container? That's IMO the nicest UX for development.

**Review by jlebon — APPROVED** [review]
Some comments, but LGTM overall!

File: docs/development/quickstart.md (hunk 1 above)
Comment (lucab) [reply docs/development/quickstart.md:1]: Ah, good catch, I totally forgot about front-matters here.

**Comment by lucab** [comment]
@jlebon @travier thanks for the feedback, I've pushed a fixup commit on top, PTAL.

**Review by travier — APPROVED** [review]
LGTM

**Review by jlebon — APPROVED** [review]

**Pull request merged**